app/src/main/java/com/example/playlistmaker/player/presentation/view_model/MusicPlayerViewModel.kt

```diff
@@ -25,41 +28,32 @@ class MusicPlayerViewModel : ViewModel() {
     val playerState: LiveData<PlayerState>
         get() = _playerState
 
-    private val mainThreadHandler by lazy { Handler(Looper.getMainLooper()) }
-
     private var playerPosition: Long = 0L
 
     fun setPlayerPosition() {
         playerPosition = mediaPlayer.currentPosition.toLong()
     }
 
-    private fun createUpdateTimerMusicTask(): Runnable {
-        return object : Runnable {
-            override fun run() {
-
-                if (playerPosition < MUSIC_TIME) {
+    private fun createUpdateTimerMusicTask() {
+        timerJob = viewModelScope.launch {
```

Comment:
При запуске новой корутины лучше отменить предыдущую Job, на всякий случай:
```suggestion
        timerJob?.cancel()
        timerJob = viewModelScope.launch {
```